app/controllers/APIController.scala

```diff
@@ -22,9 +26,52 @@ class APIController @Inject() (
     status.asJson.spaces2
   }
 
+  def parseFromJson(request: Request[AnyContent]): Option[ProductFromJson] = {
```

Comment:
A good amount of the complications you are having are because the library and the framework you are using are not interacting nicely. Try to get inspiration from this play framework tutorial and see if there are different imports that could help you:

https://dvirf1.github.io/play-tutorial/posts/circe/

Reply:
The key concept to  understand here is the concept of "custom body parsers" that the framework has

Reply:
@BenjaminGarza  were you able to read how to better integrate the play framework and circe?